python/pyspark/ml/regression.py

```diff
@@ -743,6 +743,18 @@ def treeWeights(self):
         """Return the weights for each tree"""
         return list(self._call_java("javaTreeWeights"))
 
+    @property
```

Comment:
Yeah - may as well leave it for now. Perhaps later it may make sense to
refactor it into one of the traits for Scala and Python

On Thu, 5 May 2016 at 17:22, Seth Hendrickson <notifications@github.com>
wrote:

> In python/pyspark/ml/regression.py
> <https://github.com/apache/spark/pull/12919#discussion_r62202569>:
>
> > @@ -743,6 +743,18 @@ def treeWeights(self):
> >          """Return the weights for each tree"""
> >          return list(self._call_java("javaTreeWeights"))
> >
> > +    @property
>
> This question applies to Scala API as well. Partly, I think, because in
> the Scala API things were done incrementally. But also because there may be
> some hope in the future that more predictors provide feature importances,
> and then it could be an abstract method in the Predictor class instead. I
> don't know that there's a better reason.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12919/files/6400e9e3d7e096b05475cca548e31aabca8cbaff#r62202569>
>